jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/Server.java

```diff
@@ -307,6 +355,40 @@ public String getProtocolProviderPackages() {
 		return protocolProviderPackages;
 	}
 
+	/**
+	 * Factory to create a JMXServiceURL from a pid. Inner class to prevent class
+	 * loader issues when tools.jar isn't present.
+	 */
+	private static class JMXServiceURLFactory {
+
+		public static JMXServiceURL extractJMXServiceURLFromPid(String pid) throws IOException {
```

Comment:
I'm always confused by classloaders, but in this case, it seems that we will still throw a ClassNotFoundException in case we load this on a non SUN JVM. It should run fine as long as we do not try to attach to a local process though.

I'm still thinking about this one. It seems that it does not break backward compatibility, so that's probably good enough to merge.

Any idea ?